app/src/main/java/com/onboarding/calculatorkotlin/MainActivity.kt

```diff
@@ -2,10 +2,44 @@ package com.onboarding.calculatorkotlin
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import com.onboarding.calculatorkotlin.databinding.ActivityMainBinding
+import com.onboarding.calculatorkotlin.mvp.CalculatorContract
+import com.onboarding.calculatorkotlin.mvp.model.CalculatorModel
+import com.onboarding.calculatorkotlin.mvp.presenter.CalculatorPresenter
+import com.onboarding.calculatorkotlin.mvp.view.CalculatorView
 
 class MainActivity : AppCompatActivity() {
+
+    private lateinit var presenter: CalculatorContract.Presenter
+    private lateinit var binding: ActivityMainBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        presenter = CalculatorPresenter(CalculatorModel(), CalculatorView(this, binding))
+        setOnClickListeners()
+    }
+
+    private fun setOnClickListeners() {
+        binding.buttonAdd.setOnClickListener { presenter.onOperatorButtonPressed(binding.buttonAdd.text.toString()) }
+        binding.buttonSubtraction.setOnClickListener { presenter.onMinusButtonPressed() }
+        binding.buttonMultiply.setOnClickListener { presenter.onOperatorButtonPressed(binding.buttonMultiply.text.toString()) }
+        binding.buttonDivision.setOnClickListener { presenter.onOperatorButtonPressed(binding.buttonDivision.text.toString()) }
+        binding.buttonZero.setOnClickListener { presenter.onNumberButtonPressed(binding.buttonZero.text.toString()) }
+        binding.buttonOne.setOnClickListener { presenter.onNumberButtonPressed(binding.buttonOne.text.toString()) }
```

Comment:
I think you can do a better solution to avoid calling a lot of time presenter.onNumberButtonPressed

Reply:
Solved!